mainpage/wlansi/static/wlansi/js/map.js

```diff
@@ -23,30 +26,47 @@ $(document).ready(function () {
         'mapTypeId': google.maps.MapTypeId.ROADMAP,
     };
     var map = new google.maps.Map(document.getElementById('map_canvas'), map_options);
-    $.each(nodes, function (i, node) {
-        var marker = new google.maps.Marker({
-            'position': new google.maps.LatLng(node.lat, node.long),
+
+    function updateStatus() {
+      var bounds = map.getBounds();
+      var visible_nodes = 0;
+      $.each(nodes, function (i, node) {
+        if (node.hasOwnProperty("l") && bounds.contains(new google.maps.LatLng(node.l[1], node.l[0]))) visible_nodes++;
+      });
+      var fmts = ngettext("%(visible_nodes)s active node visible (%(all_nodes)s all)", "%(visible_nodes)s active nodes visible (%(all_nodes)s all)", visible_nodes);
+      $('.map_status').text(interpolate(fmts, {
+         'visible_nodes': visible_nodes,
+         'all_nodes': nodes.length
+      }, true));
+    }
+
+    function displayNodes() {
+      $.each(nodes, function (i, node) {
+        if (node.hasOwnProperty("l")) {
+          var marker = new google.maps.Marker({
+            'position': new google.maps.LatLng(node.l[1], node.l[0]),
             'map': map,
             'shadow': shadow,
             'icon': icon,
             'shape': shape,
-            'title': node.name
-        });
-        google.maps.event.addListener(marker, 'click', function () {
-            document.location = node.url;
-        });
-    });
-    function updateStatus() {
-        var bounds = map.getBounds();
-        var visible_nodes = 0;
-        $.each(nodes, function (i, node) {
-            if (bounds.contains(new google.maps.LatLng(node.lat, node.long))) visible_nodes++;
-        });
-        var fmts = ngettext("%(visible_nodes)s active node visible (%(all_nodes)s all)", "%(visible_nodes)s active nodes visible (%(all_nodes)s all)", visible_nodes);
-        $('.map_status').text(interpolate(fmts, {
-            'visible_nodes': visible_nodes,
-            'all_nodes': nodes.length
-        }, true));
+            'title': node.n
+          });
+          google.maps.event.addListener(marker, 'click', function () {
+            document.location = nodewatcher_url + "/node/" + node.i + "/";
+          });
+        }
+      });
+      updateStatus();
     }
+
+    $.get(nodewatcher_url + "/api/v1/stream/?format=json&tags__module=topology&limit=1", function (data) {
```

Comment:
All our code is continuously evolving. :-) We all learn all the time new stuff. And then when we notice old code around, we can fix it.